### ISSUE TITLE
Discriminator with Reference to enum

### ISSUE THREAD

**Issue by DevDuck** [Rebilly/ReDoc#223]
I'm new to `discriminator` and `x-extendedDiscriminator` so let me know if I'm using it incorrectly.

The generated `select` dropdown has no options when my `discriminator` is a reference to an enum definition. My workaround is to use `type: string` but then I lose the fact that it is an `enum` in the response schema.

For example, tweaking the example given in #217:

```yaml
definitions:

  Payment:
    x-extendedDiscriminator: type
    type: object
    required:
      - type
    properties:
      type:
        $ref: '#/definitions/PaymentType'
        #type: string <-- this works, as expected; the above does not

  CashPayment:
    allOf:
      - $ref: "#/definitions/Payment"
      - properties:
          type:
            type: string
            enum:
              - cash
          currency:
            type: string

  PayPalPayment:
    allOf:
      - $ref: "#/definitions/Payment"
      - properties:
          type:
            type: string
            enum:
              - paypal
          userEmail:
            type: string

  PaymentType:
    type: string
    format: enum
    enum:
      - cash
      - paypal
      - card
``` 

So if I use the reference, then I only get an empty `select`. But if I use "`type: string`" instead of the reference, I lose:
- `PaymentType (string) <enum> / Valid values: "cash" "paypal" "card"` text in the response schema
- `"type": "string"` instead of the default `"type": "cash"` in the response samples

One other oddity I've noticed--if I use the child types (`PayPalPayment` or `CashPayment`) in a response, it'll show up with an empty `select` as well.

**Comment by RomanGotsiy** [comment]
hey @DevDuck, sorry for slow response.

Right now x-extendedDiscriminator works not as intended because of #217.
I will push a fix for that issue today.

This won't fix the issue you submit, though.

1) the way how discriminator works according to spec leads to such errors:
> When used, the discriminator will be the name of the property used to decide which schema definition is used to validate the structure of the model.

so when e.g. `cash` is selected ReDoc just renders `CashPayment` definition. It should inherit type enum and format from parent via `allOf` but I haven't yet implemented such deep `allOf` merging. I merge only properties and required at the moment. Merging internal schema is not trivial :(

so you'll be able to workaround this somehow like:

```yaml
CashPayment:
    allOf:
      - $ref: "#/definitions/Payment"
      - properties:
          type:
            allOf:
              - $ref: '#/definitions/PaymentType'
            enum:
              - cash
          currency:
            type: string
```

Next, ReDoc intentionally doesn't show `Valid values` for field with discriminators as all valid values should be in the dropdown according to the spec:
> The value of the chosen property has to be the friendly name given to the model under the definitions property.

I will let you know once #217 is fixed and we will discuss then

**Comment by RomanGotsiy** [comment]
@DevDuck a new version of ReDoc `v1.11.0` has just been released.

You can now try your example on the latest version taking into account the example of workaround I left above.
Waiting for your feedback :)

**Comment by DevDuck** [comment]
Confirmed that your fix does fix the naming issue described in #217. As well, it fixes the empty dropdown issue when using the derived types.

The workaround does indeed reflect the enum now: `PaymentType (string) <enum>` instead of just `string`. But the response sample still just shows `string` for the `type` property instead of choosing one of the enum values, when referencing the parent type in a response. (When a derived type is referenced, it correctly populates the right value.)

Thanks for the update! You're doing great work!

**Comment by RomanGotsiy** [comment]
Closing this issue as outdated.
If this issue still occurs in redoc@2.0.0-alpha.x open a new issue please.
Thanks!
